gadget/update.go

```diff
@@ -1474,9 +1495,14 @@ func canUpdateStructure(fromVss []VolumeStructure, fromIdx int, toVss []VolumeSt
 		if !from.HasFilesystem() {
 			return fmt.Errorf("cannot change a bare structure to filesystem one")
 		}
-		if from.Filesystem != to.Filesystem {
-			return fmt.Errorf("cannot change filesystem from %q to %q",
-				from.Filesystem, to.Filesystem)
+		if !fromV.HasPartial(PartialFilesystem) {
+			if toV.HasPartial(PartialFilesystem) {
+				return fmt.Errorf("changing from defined to undefined filesystem is not allowed")
+			}
+			if from.Filesystem != to.Filesystem {
+				return fmt.Errorf("cannot change filesystem from %q to %q",
+					from.Filesystem, to.Filesystem)
+			}
```

Comment:
same issues here, partial is a permission not to set something but if it is set it still needs to be checked, but maybe I'm missing something